CRM/Core/Form.php

```diff
@@ -1249,13 +1254,22 @@ public function addField($name, $props = array(), $required = FALSE) {
         if (empty($props['multiple'])) {
           $options = array('' => $props['placeholder']) + $options;
         }
-        $this->add('select', $name, $label, $options, $required, $props);
+        $props += array('required' => $required);
+        $this->addElement('select', $name, $label, $options, $props);
         // TODO: Add and/or option for fields that store multiple values
         break;
 
       //case 'AdvMulti-Select':
       //case 'CheckBox':
-      //case 'File':
+      case 'File':
+        // We should not build upload file in search mode.
+        if (isset($props['context']) && $props['context'] == 'search') {
+          return;
+        }
+        $this->addElement('file', $name, $label, $props, $required);
```

Comment:
I think we need to switch `addElement` to `add` so the required rule will be properly set.